sideEvent.c

```diff
@@ -0,0 +1,112 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include "game_data.h"
+#include "combat.h" 
+void handleSideEvent(Game* game) {
+    int eventType = rand() % 3;
+    switch (eventType) {
+        case 0:
+            handleTrapEvent(game);
+            break;
+        case 1:
+            handleChestEvent(game);
+            break;
+        case 2: 
+            handleMerchantQuest(game);
+            break;
+    }
+}
+void handleTrapEvent(Game* game) {
+    printf("You trigger a hidden trap!\n");
+    printf("1. Attempt to disarm (Rogue skill)\n");
+    printf("2. Take the damage\n");
+    printf("Choice: ");
+    int trapChoice;
+    scanf("%d", &trapChoice);
+    if (trapChoice == 1) {
+        int disarmChance = 50; 
+        if (rand() % 100 < disarmChance) {
+            printf("Successfully disarmed the trap!\n");
+            printf("You gain 15 EXP for your skill.\n");
```

Comment:
should add the stat directly here like game->champion->xp += 15 or st. please do the same for the damage